go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -139,14 +135,29 @@ func (st *symtab) Find(col *sqlparser.ColName, autoResolve bool) (rb *route, isL
 	}
 	if len(st.Colsyms) != 0 {
 		name := sqlparser.String(col)
+		var cursym *colsym
+		for _, colsym := range st.Colsyms {
+			if colsym.Alias.EqualString("*") {
+				col.Metadata = colsym
+				return colsym.Route(), true, nil
+			}
+			if colsym.Alias.EqualString(name) {
+				if cursym != nil {
+					return nil, false, fmt.Errorf("ambiguous symbol reference: %v", sqlparser.String(col))
+				}
+				cursym = colsym
+				col.Metadata = colsym
+			}
+		}
+		if cursym != nil {
+			return cursym.Route(), true, nil
+		}
 		starname := sqlparser.String(&sqlparser.ColName{
 			Name:      sqlparser.NewColIdent("*"),
 			Qualifier: col.Qualifier,
 		})
 		for _, colsym := range st.Colsyms {
-			if colsym.Alias.EqualString(name) || colsym.Alias.EqualString(starname) ||
-				colsym.Alias.EqualString("*") || colsym.ExprName.EqualString(name) ||
-				colsym.ExprName.EqualString(starname) || colsym.ExprName.EqualString("*") {
+			if colsym.QualifiedName.EqualString(name) || colsym.QualifiedName.EqualString(starname) {
 				col.Metadata = colsym
```

Comment:
Just wanted to check why we have used a comaprison of (*) in colsym.Alias and used starname in colsym.QualifiedName? why can;t we use starName only for the comparison in both?